File: tests/python/cases/case_185_thread_handoff_late_09.py
"""
Task case 185: thread_handoff_late_09 false-positive/false-negative stress pattern.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_185_thread_handoff_late_09(input_data):
    task_name = 'thread_handoff_late_09'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'stress',
        'stage': 'evaluation',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    import threading
    # ESCAPE: payload is captured by background thread closure.
    threading.Thread(target=lambda: RETAINED_AUDIT.append(payload), daemon=True).start()
    return 'ok'

File: tests/nodejs/cases/case_185_thread_handoff_late_09.js
/**
 * Task case 185: thread_handoff_late_09 false-positive/false-negative stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case185ThreadHandoffLate09(input) {
  const taskName = 'thread_handoff_late_09';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'stress',
    stage: 'evaluation',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: async microtask captures payload beyond return edge.
  queueMicrotask(() => retainedAudit.push(payload));
  return 'ok';
}

module.exports = {
  case185ThreadHandoffLate09,
};
